tasks/shell.js

```diff
@@ -118,6 +119,10 @@ module.exports = function( grunt ) {
                 } else if(options.stdout === true || grunt.option('verbose')) {
                     log.write(data);
                 }
+
+                if (shouldWaitUntilReady && options.readyRegex.test(data)) {
```

Comment:
What guarantees does stdout.on make regarding how data is delivered here?  Could it be delivered on character at a time?  One line at a time?  How do we know the regex will be possible if two chunks come through?